src/TestData/Factories/DonationFactory.php

```diff
@@ -97,6 +102,40 @@ public function getDonationCurrency() {
 		return $this->currency;
 	}
 
+
+	/**
+	 * @param string $date
+	 */
+	public function setDonationStartDate( $date ) {
+		$this->startDate = $date;
+	}
+
+	/**
+	 * @return string
+	 */
+	public function getDonationDate() {
+		if ( ! $this->isValidDate( $this->startDate ) ) {
```

Comment:
Instead of validating the start date in the getter, I would suggest validating the value when it is set. Then throw a warning.

Reply:
Resolved cde0350